02-hey-jupyter/README.md

```diff
@@ -0,0 +1,151 @@
+
+## Key Concepts
+
+1. **Cells**: Jupyter organizes documents into _cells_, which can contain
+   rich text in [Markdown](https://en.wikipedia.org/wiki/Markdown) format,
+   code, or the result of executed code. When a code-containing cell is
+   executed via the _Run Cell_ command, the input and output are assigned
+   a execution order indicator.
+   
+   For example, a cell marked marked `In [20]` means that the cell was last
+   executed _after_ any cell numbered 19 or less, and _before_ any cell
+   numbered 20 or more. It is often helpful to use the **Cell > Run All**
+   menu option to ensure that all cells are up-to-date.
+
+1. **Data Frames**: _pandas_ manages data in a tabular format called Data
+   Frames, where the rows define individual records and the columns define
+   various properties of the record (for example, a Person record might
+   have columns labelled Name, Date of Birth, Birth Location, etc.) Input
+   data is first read into data frames, and a set of built-in functions
+   can be used to sort, filter, or otherwise process the data as needed.
+
+## Exercises
+
+### Hello World
+
+Start the Jupyter Notebook server. Depending on your installation method,
+there may be a "Jupyter Notebook" application, or you can open a terminal
+(such as Command Prompt on Windows) and run `jupyter notebook`.
+  
+Running the command should open your default browser and navigate to the
+running Jupyter instance on your local system, but if it doesn't, the
+system should print a special URL to copy/paste into your browser:
+   
+```
+$ jupyter notebook
+[I 10:18:00.711 NotebookApp] Serving notebooks from local directory: C:\projects\training\02-hey-jupyter
+[I 10:18:00.711 NotebookApp] 0 active kernels
+[I 10:18:00.711 NotebookApp] The Jupyter Notebook is running at:
+[I 10:18:00.711 NotebookApp] http://localhost:8888/?token=3635b10886fafde18eb8dc781639b068b00299b99ff76aa4
+[I 10:18:00.711 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
+[C 10:18:00.714 NotebookApp]
+
+    Copy/paste this URL into your browser when you connect for the first time,
+    to login with a token:
+        http://localhost:8888/?token=3635b10886fafde18eb8dc781639b068b00299b99ff76aa4
+[I 10:18:00.987 NotebookApp] Accepting one-time-token-authenticated connection from ::1
+```
+
+The interface should show a list of all files in this directory.
+
+### My First Notebook
+
+Open the `scigrid.ipynb` file and execute all the cells. The CSV files
+containing information about terminal nodes (such as substations) and
+transmission lines should appear.
+
+> Aside: The .ipynb extension refers to IPython, or Interactive Python,
+> a predecessor to Jupyter.
+
+### Count Records
+
+Let's determine the number of substations in this dataset. Filter the
+`terminals` data frame for entries with a `typ` of "substation":
+
+```
+substations = terminals[terminals["typ"] == "substation"]
+substations
```

Comment:
I recommend getting in the habit of printing the head of data frames. Especially data frames of unknown size
`substations.head()`